pandas/tools/tests/test_merge.py

```diff
@@ -261,6 +261,20 @@ def test_join_on_fails_with_different_column_counts(self):
                             index=tm.makeCustomIndex(10, 2))
             merge(df, df2, right_on='a', left_on=['a', 'b'])
 
+    def test_join_on_fails_with_left_series(self):
+        with tm.assertRaisesRegexp(ValueError, 'Series'):
+            series = Series([0, 1, 2])
```

Comment:
iterate thru a bunch of things (e.g. ndarray, number, string) and assert that they raise as well